lib/errors/oauth-error.js

```diff
@@ -0,0 +1,34 @@
+
+/**
+ * Module dependencies.
+ */
+
+var StandardHttpError = require('standard-http-error');
+var util = require('util');
+
+/**
+ * Constructor.
+ */
+
+function OAuthError(messageOrError, properties) {
+  var message = messageOrError instanceof Error ? messageOrError.message : messageOrError;
+  var error = messageOrError instanceof Error ? messageOrError : null;
+
+  if (error) {
+    properties.inner = error;
```

Comment:
This class is not intended to be subclassed by users of the _node-oauth-server_ module. It is meant to be subclassed by other errors in this module.

I can coalesce `properties` to an empty object if you feel it is more correct but I normally operate under the assumption that we shouldn't defend against programmer errors and prefer to do so by implementing tests.